pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -342,12 +342,18 @@ func (s *discovery) getServiceInfo(proc *process.Process) (*serviceInfo, error)
 		return nil, err
 	}
 
+	cwd, err := proc.Cwd()
```

Comment:
my concern here is that proc.Cwd() actually allocates a string, which could be pretty long and we do it for any handled process, while we use it only for "puma" applications. What do you think?

Reply:
Pushed a proposed change to store the pointer to the Process instance instead of the cwd, and only calling the Cwd() method in the Rails detector.